### ISSUE TITLE
Loader API

### ISSUE THREAD

**Issue by gbj** [gbj/leptos#14]
Loaders should probably be something more like the following. This allows them to do fine-grained reactive reloads only in response to particular params or queries on which they actually depend. Currently, they do a Remix-style “reload whenever query or params change,” which is not great. The main issue here is establishing a stable URL for the `create_server_resource` (or whatever) so that the loader runs on either client or server, but the server resource runs only on server and can be accessed using a JSON endpoint.

```rust
fn contact_data(cx: Scope) -> Resource<String, Contact>{
  let params = use_params_map(cx);
  let id = create_memo(cx, |_| params.get("id"));
  create_server_resource(cx, move || id(), |id: String| fetch_contact_data_for_user(&id))
}
```

**Comment by gbj** [comment]
I'll be doing some work on this over on the [`loader`](https://github.com/gbj/leptos/tree/loader) branch if you're interested in following along.

**Comment by gbj** [comment]
Okay, my work on this is moving over to [`server-rpc`](https://github.com/gbj/leptos/tree/server-rpc)

So far this is just a manual implementation but I'm actually pretty excited about it... What it's showing right now is that ability to run arbitrary code on the server and have it reactively update the client. It's just done completely manually.

I'll be building out a proc macro etc.

**Comment by gbj** [comment]
Closing this now that I've merged #40. I still need to add some more docs but I'm too tired tonight!
